res/controllers/Numark_DJ2GO2_Touch_scripts.js

```diff
@@ -0,0 +1,214 @@
+            break;
+        case 0x9F: // Push.
+            this.onButtonEvent(value);
+        }
+    }
+});
+
+DJ2GO2Touch.masterGain = new components.Pot({
+    midi: [0xBF, 0x0A],
+    group: "[Master]",
+    key: "gain"
+});
+
+DJ2GO2Touch.cueGain = new components.Pot({
+    midi: [0xBF, 0x0C],
+    group: "[Master]",
+    key: "headGain"
+});
+
+DJ2GO2Touch.crossfader = new components.Pot({
+    midi: [0xBF, 0x08],
+    group: "[Master]",
+    key: "crossfader"
+});
+
+// implement a constructor for a custom Deck object specific to your controller
+DJ2GO2Touch.Deck = function(deckNumbers, midiChannel) {
+    // Call the generic Deck constructor to setup the currentDeck and deckNumbers properties,
+    // using Function.prototype.call to assign the custom Deck being constructed
+    // to 'this' in the context of the generic components.Deck constructor
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call
+    components.Deck.call(this, deckNumbers);
+
+    this.playButton = new components.PlayButton([0x90 + midiChannel, 0x00]);
+    this.cueButton = new components.CueButton([0x90 + midiChannel, 0x01]);
+    this.syncButton = new components.SyncButton([0x90 + midiChannel, 0x02]);
+
+    this.pflButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x1B],
+        key: "pfl"
+    });
+
+    this.loadButton = new components.Button({
+        midi: [0x9F, 0x02 + midiChannel],
+        key: "LoadSelectedTrack",
+        input: function(channel, control, value, status, _group) {
+            this.send(this.isPress(channel, control, value, status) ? this.on : this.off);
+            components.Button.prototype.input.apply(this, arguments);
+        }
+    });
+
+    this.preGain = new components.Pot({
+        midi: [0xB0 + midiChannel, 0x16],
+        group: "[QuickEffectRack1_" + this.currentDeck + "]",
+        key: "super1"
+    });
+
+    engine.setValue(this.currentDeck, "rate_dir", -1);
+    this.tempoFader = new components.Pot({
+        group: "[Channel" + deckNumbers + "]",
+        midi: [0xB0 + midiChannel, 0x09],
+        key: "rate",
+        inSetParameter: components.Pot.prototype.inSetParameter,
+        connect: function() {
+            engine.softTakeover(this.group, "rate", true);
+            components.Pot.prototype.connect.apply(this, arguments);
+        }
+    });
+
+    this.hotcueButtons = [];
+    this.samplerButtons = [];
+    this.beatloopButtons = [];
+    for (var i = 1; i <= 4; i++) {
+        this.hotcueButtons[i] = new components.HotcueButton({
+            group: "[Channel" + deckNumbers + "]",
+            midi: [0x94 + midiChannel, 0x01 + i],
+            number: i
+        });
+        this.sampler = i + (midiChannel * DJ2GO2Touch.PAD_SIZE);
+        this.samplerButtons[i] = new components.SamplerButton({
+            group: "[Sampler" + this.sampler + "]",
+            midi: [0x94 + midiChannel, 0x31 + i],
+            number: this.sampler
```

Comment:
making it part of the object is overkill. Just make it a local variable
```suggestion
        var sampler = i + (midiChannel * DJ2GO2Touch.PAD_SIZE);
        this.samplerButtons[i] = new components.SamplerButton({
            group: "[Sampler" + sampler + "]",
            midi: [0x94 + midiChannel, 0x31 + i],
            number: sampler
```